wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/SpecialNames.scala

```diff
@@ -26,4 +26,29 @@ object SpecialNames {
 
   /** A unique simple method name to map all method *signatures* into `MethodName`s. */
   val normalizedSimpleMethodName = SimpleMethodName("m")
+
+  // Memory Instructions
+  val WasmMemoryAllocatorClass = ClassName("scala.scalajs.wasm.MemoryAllocator")
+  val WasmMemorySegmentClass = ClassName("scala.scalajs.wasm.MemorySegment")
+  val WasmMemoryAllocatorCtor = MethodName.constructor(Nil)
+  val WasmMemorySegmentCtor = MethodName.constructor(List(IntRef, IntRef))
+
+  val WasmMemorySegmentStartAddressField =
+    FieldName(WasmMemorySegmentClass, SimpleFieldName("start"))
+  val WasmMemorySegmentSizeField = FieldName(WasmMemorySegmentClass, SimpleFieldName("size"))
+  val allocateMethodName =
+    MethodName("allocate", List(IntRef), ClassRef(SpecialNames.WasmMemorySegmentClass))
+  val freeMethodName = MethodName("free", Nil, VoidRef)
+
+  val loadByteMethodName = MethodName("_loadByte", List(IntRef), IntRef)
+  val loadIntMethodName = MethodName("_loadInt", List(IntRef), IntRef)
+  val loadMethodNames = Set(loadByteMethodName, loadIntMethodName)
+
+  val storeByteMethodName = MethodName("_storeByte", List(IntRef, ByteRef), VoidRef)
+  val storeIntMethodName = MethodName("_storeInt", List(IntRef, IntRef), VoidRef)
+  val storeMethodNames = Set(storeIntMethodName, storeByteMethodName)
+
+  // WASI functions
+  val WASI = ClassName("scala.scalajs.wasm.wasi$")
+  val wasiFdWrite = MethodName("fdWrite", List(IntRef, IntRef, IntRef, IntRef), IntRef)
```

Comment:
These we should find a way to make generic. The backend should know about memory instructions, because they are opcodes. But library functions coming from imports shouldn't be hard-coded. We should make it possible for *users* to define these interoperability anchors as library themselves.

It might be hard/impossible to do as long as we're not merged upstream, though. We would likely do this based on some sort of annotation (like `@WasmImport`), but annotations are not persisted in the IR in general.